railties/lib/rails/tasks/routes.rake

```diff
@@ -1,7 +1,28 @@
+require 'optparse'
+
 desc 'Print out all defined routes in match order, with names. Target specific controller with CONTROLLER=x.'
 task routes: :environment do
   all_routes = Rails.application.routes.routes
   require 'action_dispatch/routing/inspector'
   inspector = ActionDispatch::Routing::RoutesInspector.new(all_routes)
-  puts inspector.format(ActionDispatch::Routing::ConsoleFormatter.new, ENV['CONTROLLER'])
+
+  if ENV['CONTROLLER']
+    ActiveSupport::Deprecation.warn('`CONTROLLER` environment is deprecated and will be removed in Rails 5.1. Please use -g and -c keys instead.')
```

Comment:
I feel we need to explain the controller ENV variable better. It would be great to give up an upgrade example too.

How about:

```ruby
# At the top:
require 'active_support/deprecation'
require 'active_support/core_ext/string/strip' # for strip_heredoc

# ...

ActiveSupport::Deprecation.warn <<-eow.strip_heredoc
  Passing `CONTROLLER` to `bin/rake routes` is deprecated and will be removed in Rails 5.1.
  Please use `bin/rake routes --controller #{ENV['CONTROLLER']} instead.
eow
```